### PR TITLE
Fix scapy sdist

### PR DESCRIPTION
**Pull request by evverx**
In 669506bd42e4141718374ba297974881fe125fb8 scapy switched to
pyproject.toml where 'readme' is dynamic and should be supplied by
setup.py. setup.py reads README.md and it works from inside the source
tree but since README.md isn't included in the sdist it fails when pip
or the build frontend builds the wheel from the sdist with:
```
distutils.errors.DistutilsOptionError: No configuration found for dynamic 'readme'.
Some dynamic fields need to be specified via `tool.setuptools.dynamic`
others must be specified via the equivalent attribute in `setup.py`.
```

This patch addresses that by removing README and letting setuptools
add README.md to the sdist automatcially.

README was added in 4f71027fcdcb1e4ca8d7e04987588d2f7e615f4b in 2016 and back then it probably made
sense because setuptools didn't include README.md in the sdist
automatically. These days setuptools can handle README.md just fine so
it's no longer necessary to keep README any more.

It makes it possible to build the wheel from the sdist again.

Fixes:
```
python3 -m build
...
* Building wheel from sdist
* Creating venv isolated environment...
* Installing packages in isolated environment... (setuptools>=62.0.0)
* Getting build dependencies for wheel...
...
  File "/tmp/build-env-vtaxy4xl/lib64/python3.11/site-packages/setuptools/config/pyprojecttoml.py", line 351, in _obtain_readme
    self._ensure_previously_set(dist, "readme")
  File "/tmp/build-env-vtaxy4xl/lib64/python3.11/site-packages/setuptools/config/pyprojecttoml.py", line 307, in _ensure_previously_set
    raise OptionError(msg)
distutils.errors.DistutilsOptionError: No configuration found for dynamic 'readme'.
Some dynamic fields need to be specified via `tool.setuptools.dynamic`
others must be specified via the equivalent attribute in `setup.py`.

ERROR Backend subprocess exited when trying to invoke get_requires_for_build_wheel
```
and
```
python3 -m pip install dist/scapy-2.5.0.dev56.tar.gz
Processing ./dist/scapy-2.5.0.dev56.tar.gz
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
...
```

It's a follow-up to 669506bd42e4141718374ba297974881fe125fb8

### CODE REVIEW

File: .github/workflows/unittests.yml [hunk 1]
```diff
@@ -29,6 +29,8 @@ jobs:
         run: tox -e spell
       - name: Run twine check
         run: tox -e twine
+      - name: Run build check
+        run: tox -e build
```
Comment (evverx) [inline .github/workflows/unittests.yml:33]: FWIW I tested it without this patch and it failed as expected: https://github.com/evverx/scapy/actions/runs/4452762824/jobs/7820708250

File: setup.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ def process_ignore_tags(buffer):
             x for x in buffer.split("\n") if "<!-- ignore_ppi -->" not in x
         )
     try:
-        fpath = os.path.join(os.path.dirname(__file__), "README.md")
+        fpath = os.path.join(os.path.dirname(__file__), "README")
```
Comment (evverx) [inline setup.py:30]: Personally I'd prefer to keep `README.md` here and just remove `README` from the repository. Looking at 4f71027fcdcb1e4ca8d7e04987588d2f7e615f4b it seems it was added in 2016 and back then `setuptools` didn't add `README.md` to sdists automatically. These days `setuptools` can handle `README.md` just fine if `README` isn't present. Removing `README` can in theory break something (even though in practice it shouldn't) so this patch is somewhat more backward-compatible.

Comment (evverx) [reply setup.py]: I went ahead and removed README.

**Comment by gpotter2** [comment]
Thanks a lot for the PR !
I tested it and it looks almost good. I'm just having issues with `scapy/VERSION` not being generated when using this.
I am starting to think that we might need something like this https://setuptools.pypa.io/en/latest/build_meta.html#dynamic-build-dependencies-and-other-build-meta-tweaks instead of the https://github.com/secdev/scapy/blob/master/setup.py#L40 thing

**Comment by evverx** [comment]
Let me double-check. I unpacked the sdist and the wheel to make sure VERSION was there:
```
tar zxvf dist/scapy-2.5.0.dev59.tar.gz
cat scapy-2.5.0.dev59/scapy/VERSION
2.5.0.dev59
```
```
unzip -d WHEEL dist/scapy-2.5.0.dev59-py3-none-any.whl
cat WHEEL/scapy/VERSION
2.5.0.dev59
```
but it could be that I missed something. Could you show the build command where VERSION isn't generated? I'll try to figure out how to make it work.

**Comment by gpotter2** [comment]
If you delete `scapy/VERSION` from the root and call `python -m build`, I personally get an error

**Comment by evverx** [comment]
I can't seem to reproduce it locally
```sh
$ git clean -dxff
$ ls scapy/VERSION
ls: cannot access 'scapy/VERSION': No such file or directory
$ python -m build
...
Successfully built scapy-2.5.0.dev59.tar.gz and scapy-2.5.0.dev59-py3-none-any.whl
```
The CI can't seem to reproduce it either. I'll try to downgrade stuff to see if it helps to reproduce it.

**Comment by evverx** [comment]
I think I got it. The following warning
```
warning: no files found matching 'scapy/VERSION'
```
pops up with and without this PR when scapy is built (with both `build` and `setup.py sdist`). I didn't notice it because the build didn't fail because of that.

**Comment by evverx** [comment]
Looks like this warning was introduced in 915fe2d207b0f91768de2a25d576767a4cf2efaf
> Do not try to generate the scapy/VERSION file when importing scapy
anymore but generate it by overriding the sdist command in setup.py and
write it to the temp folder used for the source archive generation

and since `scapy/VERSION` is included in `MANIFEST.md` the warning is generated. It should be possible to "fix" it by removing it from `MANIFEST.md` but it would break the wheel in the sense that `VERSION` would no longer be included in the wheel generated from the sdist. @rjarry could you take a look? I can't say I fully understand all the corner cases there and I don't want to break it accidentally.

**Comment by evverx** [comment]
I opened https://github.com/secdev/scapy/issues/3955 but I think it's a separate issue and it would probably be easier to fix it with the "pip install <sdist>" scenario fixed.

**Comment by evverx** [comment]
Closing in favor of https://github.com/secdev/scapy/pull/3958

**Pull request closed without merging**